src/main/java/com/github/manolo8/darkbot/gui/titlebar/ExtraButton.java

```diff
@@ -31,12 +34,44 @@
     private static final Set<ExtraMenuProvider> EXTRA_DECORATIONS = new LinkedHashSet<>();
     private static final Lazy<Void> clean = new Lazy.NoCache<>();
 
-    public static void setExtraDecorations(Stream<ExtraMenuProvider> provider) {
+    public static void setExtraDecorations(Stream<ExtraMenuProvider> provider, FeatureRegistry featureRegistry) {
         EXTRA_DECORATIONS.clear();
-        provider.forEach(EXTRA_DECORATIONS::add);
+        PluginExtraMenuProvider.PLUGINS.clear();
+
+        provider.forEach(extra -> {
+            Plugin pl = featureRegistry.getFeatureDefinition(extra).getPlugin();
+            if (pl != null && extra.autoSubmenu()) {
+                PluginExtraMenuProvider.PLUGINS.computeIfAbsent(
+                        pl,
+                        features -> new LinkedHashSet<>()).add(extra);
+            } else {
+                EXTRA_DECORATIONS.add(extra);
+            }
+        });
+
```

Comment:
You'll here need to remove the `PluginExtraMenuProvider` from EXTRA_DECORATION if it exists, and re-add it, so that it gets sorted to the end. Alternatively you could try to add a sorter to the provider stream, but i think this will be easier